helpers/checkpoint_logger/__init__.py

```diff
@@ -358,11 +360,13 @@ def __init__(
         cls: type[T],
         data: Optional[MutableMapping[T, int]] = None,
         strict=False,
+        context: CheckpointContext = None,
     ):
         self.cls = cls
         self.data = data if data else {}
         self.kwargs_key = _kwargs_key(self.cls)
         self.strict = strict
+        self.context = context if context else {}
```

Comment:
This should just be `context` in this case, as defaulting to `{}` does not really make sense.